src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php

```diff
@@ -180,6 +180,39 @@ private function get_balanced_tag_bookmarks() {
 		return array( $opener_tag, $closer_tag );
 	}
 
+	/**
+	 * Skips processing the content between tags.
+	 *
+	 * It positions the cursor in the closer tag of the foreign element, if it
+	 * exists.
+	 *
+	 * This function is intended to skip processing SVG and MathML inner content
+	 * instead of bailing out the whole processing.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @access private
+	 *
+	 * @return bool Whether the foreign content was successfully skipped.
+	 */
+	public function skip_foreign_content(): bool {
+		$depth    = 1;
+		$tag_name = $this->get_tag();
+		while ( $depth > 0 && $this->next_tag(
+			array(
+				'tag_name'    => $tag_name,
+				'tag_closers' => 'visit',
+			)
+		) ) {
+			if ( $this->has_self_closing_flag() ) {
+				continue;
+			}
+			$depth += $this->is_tag_closer() ? -1 : 1;
```

Comment:
@dmsnell What about a `title` tag that appears inside of an SVG? Since it's in XML parsing mode, would it still be processed as rcdata here?

Reply:
@westonruter this is a great question, and you're right. it's not RCDATA when found inside SVG. this is all the more reason to focus only on SVG and MathML tags and ignore the others.

in https://github.com/WordPress/wordpress-develop/pull/6006 what I'm looking to do is have the HTML Processor tell the Tag Processor that it's inside foreign content, and while that flag is active, we'll disable the special handling of tokens like TITLE and the others.

it's complicated and we explored trying to put the foreign-content handling code inside the Tag Processor, but that ended up demanding that we re-implement much of the HTML Processor's functionality there, so it will probably be the case that you need the HTML Processor to adequately parse inside SVG and MathML.

in the meantime, if one overlooks the HTML elements which break out into the nearest "integration point," then I think that skipping from SVG tag to SVG tag will be reliable enough for now.

Reply:
aha I misread your statement @westonruter concerning `TITLE`. in this case it shouldn't matter that we're checking `is_tag_closer()` because we're not visiting the `TITLE` tag - we're skipping straight to whichever tag (`SVG` or `MATH`) opened the foreign content.